src/primitives.rs

```diff
@@ -63,7 +63,7 @@ pub struct DiatomicWaker {
 
 impl DiatomicWaker {
     /// Creates a new `DiatomicWaker`.
-    pub fn new() -> Self {
+    pub const fn new() -> Self {
```

Comment:
This breaks loom (see the CI errors), I guess we could work around this with a `[cfg(not(diatomic_waker_loom))]`.